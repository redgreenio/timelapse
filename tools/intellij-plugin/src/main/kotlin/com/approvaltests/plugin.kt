package com.approvaltests

const val ID = "io.redgreen.timelapse.dev"
const val DISPLAY_NAME = "Approval Tests"
