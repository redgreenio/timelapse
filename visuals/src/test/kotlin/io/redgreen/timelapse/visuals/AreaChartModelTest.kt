package io.redgreen.timelapse.visuals

import org.approvaltests.Approvals
import org.junit.jupiter.api.Test

class AreaChartModelTest {
  @Test
  fun `increasing insertions`() {
    // given
    val side = 100
    val pointRadius = 1
    val commits = listOf(1, 2, 3, 4, 5).map(::Commit)
    val outPoints = mutableListOf<Point>()

    // when
    computePolygonPoints(commits, side, side, pointRadius, outPoints)

    // then
    Approvals.verify(outPoints.humanize())
  }

  @Test
  fun `increasing insertions in a 0x0 area with 0 point radius`() {
    // given
    val side = 0
    val pointRadius = 0
    val commits = listOf(1, 2, 3, 4, 5).map(::Commit)
    val outPoints = mutableListOf<Point>()

    // when
    computePolygonPoints(commits, side, side, pointRadius, outPoints)

    // then
    Approvals.verify(outPoints.humanize())
  }

  @Test
  fun `varying insertions`() {
    // given
    val side = 100
    val pointRadius = 5
    val commits = listOf(8, 7, 5, 4, 6).map(::Commit)
    val outPoints = mutableListOf<Point>()

    // when
    computePolygonPoints(commits, side, side, pointRadius, outPoints)

    // then
    Approvals.verify(outPoints.humanize())
  }

  @Test
  fun `negative numbers`() {
    // given
    val side = 100
    val pointRadius = 1
    val commits = listOf(-3 ,-2, -1, 0, 1, 2, 3).map(::Commit)
    val outPoints = mutableListOf<Point>()

    // when
    computePolygonPoints(commits, side, side, pointRadius, outPoints)

    // then
    Approvals.verify(outPoints.humanize())
  }

  private fun MutableList<Point>.humanize() =
    this.joinToString("\n") { "x=${it.x}, y=${it.y}" }
}
