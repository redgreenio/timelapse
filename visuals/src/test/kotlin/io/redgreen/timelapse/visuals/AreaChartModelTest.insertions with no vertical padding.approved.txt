x=0, y=100
x=20, y=80
x=40, y=60
x=60, y=40
x=80, y=20
x=100, y=0
x=100, y=100
x=0, y=100